components/provisioner/internal/provisioning/service.go

```diff
@@ -176,32 +183,43 @@ func (r *service) unregisterFailedRuntime(id, tenant string) {
 func (r *service) DeprovisionRuntime(id string) (string, apperrors.AppError) {
 	session := r.dbSessionFactory.NewReadWriteSession()
 
-	err := r.verifyLastOperationFinished(session, id)
-	if err != nil {
-		return "", err
+	appErr := r.verifyLastOperationFinished(session, id)
+	if appErr != nil {
+		return "", appErr
 	}
 
 	cluster, dberr := session.GetCluster(id)
 	if dberr != nil {
 		return "", apperrors.Internal("Failed to get cluster: %s", dberr.Error())
 	}
 
-	operation, err := r.provisioner.DeprovisionCluster(cluster, r.uuidGenerator.New())
-	if err != nil {
-		return "", apperrors.Internal("Failed to start deprovisioning: %s", err.Error())
+	operation, appErr := r.provisioner.DeprovisionCluster(cluster, r.uuidGenerator.New())
+	if appErr != nil {
+		return "", apperrors.Internal("Failed to start deprovisioning: %s", appErr.Error())
 	}
 
 	dberr = session.InsertOperation(operation)
 	if dberr != nil {
 		return "", apperrors.Internal("Failed to insert operation to database: %s", dberr.Error())
 	}
 
-	if util.IsNilOrEmpty(cluster.ActiveKymaConfigId) {
-		log.Infof("Starting deprovisioning steps for runtime %s without installation", cluster.ID)
-		r.deprovisioningNoInstallQueue.Add(operation.ID)
-	} else {
+	if cluster.Kubeconfig == nil {
+		return "", apperrors.Internal("error: kubeconfig is nil")
+	}
+
+	k8sConfig, err := k8s.ParseToK8sConfig([]byte(*cluster.Kubeconfig))
+	if err != nil {
+		return "", apperrors.Internal("error: failed to create kubernetes config from raw: %s", err.Error())
+	}
+
+	installationState, err := r.installationClient.CheckInstallationState(k8sConfig)
+
+	if err == nil && util.NotNilOrEmpty(cluster.ActiveKymaConfigId) && installationState.State == string(v1alpha1.StateInstalled) {
```

Comment:
I'm not sure about the `installationState.State == string(v1alpha1.StateInstalled)` part. For example, if it's in `StateError` there could be some problems with deleting the cluster without uninstalling Kyma first. Is that correct? (Just asking)

Reply:
After discussion changed to following:

```
if err != nil || util.IsNilOrEmpty(cluster.ActiveKymaConfigId) || installationState.State == installationSDK.NoInstallationState {
		log.Infof("Starting deprovisioning steps for runtime %s without installation", cluster.ID)
		r.deprovisioningNoInstallQueue.Add(operation.ID)
	} else {
		log.Infof("Starting deprovisioning steps for runtime %s with installation", cluster.ID)
		r.deprovisioningQueue.Add(operation.ID)
	}
```